data/generic_vector.go

```diff
@@ -52,3 +52,16 @@ func (v *genVector) Type() FieldType {
 func (v *genVector) Extend(i int) {
 	(*v) = append((*v), make([]gen, i)...)
 }
+
+func (v *genVector) InsertAt(i int, val interface{}) {
+	if v.Len() < i {
+		v.Append(val)
+	} else {
+		v.Extend(1)
+		for j := v.Len() - 1; j > i; j-- {
+			previousVal, _ := v.ConcreteAt(j - 1)
```

Comment:
I think this would be `At`, not sure why `ConcreteAt`?. Also since in this case we have access to the slice (gen code). See alternative way of Insert at https://github.com/golang/go/wiki/SliceTricks#insert